docs: Added information about how the amount of TransferSnsTreasuryFunds proposals is capped. (#2796)
dfinity/portal: docs/developer-docs/daos/sns/managing/making-proposals.mdx

This is a ["critical" proposal type](#criticalproposals), which means it is subject to higher voting thresholds before it is accepted.

#### Maximum 7-day Amount Total

> The rate that funds can be transferred from the treasury is limited. To explain how this works,

Reviewer: ```suggestion
The rate at which funds can be transferred from the treasury is limited. To explain how this works,
```

> The rate that funds can be transferred from the treasury is limited. To explain how this works,
> first, define T to be the value of the tokens in the treasury in units of [XDR]. T can fall within
> three size regimes:
> 

Reviewer: ```suggestion
The rate at which funds can be transferred from the treasury is limited by some rules.
These take into account how many tokens, measured in XDR value, are still in the treasury.
The rules distinguish between three categories that we define as T:

```

[XDR]: https://en.wikipedia.org/wiki/Special_drawing_rights

<table>
  <thead>
    <tr>
      <th>Size</th>
      <th>Max T (XDR)</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>small</td>

>       <td>100_000</td>

Reviewer: how about giving the range here (0-100_000) rather than the max? I think the max is a bit harder to parse... We mean any T that falls into this bucket, right?

    </tr>
    <tr>
      <td>medium</td>
      <td>1_200_000</td>
    </tr>
    <tr>
      <td>large</td>
      <td>(infinity)</td>
    </tr>
  </tbody>
</table>

> The total amount that can be transferred from the treasury in a 7 day period is at most L(T), where
> L(T) is given by this table:

Reviewer: ```suggestion
The total amount that can be transferred from the treasury in a 7 day period is at most L(T), where
L(T) defines the rules that depend on the amount T in the treasury and is defined by the following table.
```

Author: Done.

<table>
  <thead>
    <tr>
      <th>Treasury Size</th>
      <th>L(T)</th>
      <th>Fraction of T</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>small</td>
      <td>T</td>
      <td>100%</td>
    </tr>
    <tr>
      <td>medium</td>
      <td>0.25 * T</td>
      <td>25%</td>
    </tr>
    <tr>
      <td>large</td>
      <td>300_000 XDR</td>
      <td>0% - 25%</td>

>     </tr>

Reviewer: I don't quite understand what the two columns in the table mean. Is it the larger of the two or are these just two alternative ways how to state the same maximum?

  </tbody>
</table>

> 

Reviewer: how about adding an example for each category or at least one overall example to explain how to read the tables?

This is a ["critical" proposal type](#criticalproposals), which means it is subject to higher voting thresholds before it is accepted.

> #### Maximum 7-day Amount Total

Reviewer 2: ```suggestion
#### Maximum 7-day amount total
```

The rate at which funds can be transferred from the treasury is capped. The cap varies based on the
value of the tokens in the treasury in [XDR], denoted as T. For the purposes of capping, T can fall
within three ranges:

> [XDR]: https://en.wikipedia.org/wiki/Special_drawing_rights

Reviewer 2: ```suggestion

```

<table>
  <thead>
    <tr>
      <th>Size</th>
      <th>Range of T (XDR)</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>Small</td>
      <td>0 &le; T &le; 100_000</td>
    </tr>
    <tr>
      <td>Medium</td>
      <td>100_000 < T &le; 1_200_000</td>
    </tr>
    <tr>
      <td>Large</td>
      <td>1_200_000 < T < ♾️</td>
    </tr>
  </tbody>
</table>

The total amount that can be transferred from the treasury in a 7 day period is at most L(T), where

> L(T) defines the rules that depend on the amount T in the treasury and is defined by the following

Reviewer 2: ```suggestion
L(T) defines the rules that depend on the amount T in the treasury and is defined in the following
```

> table.

Reviewer 2: ```suggestion
table:
```

<table>
  <thead>
    <tr>

>       <th>Treasury Size</th>

Reviewer 2: ```suggestion
      <th>Treasury size</th>
```

      <th>L(T)</th>
      <th>Fraction of T</th>
    </tr>
  </thead>
  <tbody>
    <tr>

>       <td>small</td>

Reviewer 2: ```suggestion
      <td>Small</td>
```

      <td>T</td>
      <td>100%</td>
    </tr>
    <tr>

>       <td>medium</td>

Reviewer 2: ```suggestion
      <td>Medium</td>
```

      <td>0.25 * T</td>
      <td>25%</td>
    </tr>
    <tr>

>       <td>large</td>

Reviewer 2: ```suggestion
      <td>Large</td>
```

      <td>300_000 XDR</td>
      <td>0% - 25%</td>
    </tr>
  </tbody>
</table>

ICP and SNS tokens are considered separately.

For example, if the treasury contains 75_000 XDR worth of SNS tokens, then this amount is considered
"small". In this case up to 75_000 XDR worth of SNS tokens can be transferred from the treasury.

To assess the tokens in the treasury, external price information is fetched at the time of proposal
submission.

> The price of ICP is taken from the Cycles Minting Canister's

Reviewer 2: ```suggestion
The price of ICP is taken from the cycles minting canister's
```

[get_average_icp_xdr_conversion_rate][icp-price] method.

> [icp-price]: https://sourcegraph.com/github.com/dfinity/ic@8efb7e81a1f1257a38621b1f12eccc8b8b1c7e39/-/blob/rs/nns/cmc/src/main.rs?L901:33-901:44

Reviewer 2: ```suggestion

```

> The price of the SNS token in ICP is taken from the Swap Canister's [get_derived_state] method.

Reviewer 2: ```suggestion
The price of the SNS token in ICP is taken from the swap canister's [get_derived_state](https://sourcegraph.com/github.com/dfinity/ic@8efb7e81a1f1257a38621b1f12eccc8b8b1c7e39/-/blob/rs/sns/swap/proto/ic_sns_swap/pb/v1/swap.proto?L1304-1325) method.
```

> [get_derived_state]: https://sourcegraph.com/github.com/dfinity/ic@8efb7e81a1f1257a38621b1f12eccc8b8b1c7e39/-/blob/rs/sns/swap/proto/ic_sns_swap/pb/v1/swap.proto?L1304-1325

Reviewer 2: ```suggestion

```
